### PR TITLE
trigger: doc overhaul

### PR DESCRIPTION
**Pull request by dgw**
This is pretty self-explanatory. I'm opening it early to get feedback on my approach to documenting `PreTrigger` (which wasn't really documented before).

Note I'm still working over the existing docstrings in `Trigger` as of PR opening. Y'all can expect force-pushes to this branch as I find more time to work on these updates.

I expect this to be the last major documentation overhaul I do before 7.0, especially since I'm leaving the country for two weeks later this month; the few remaining submodules in #1565 are much less important (either recently written or purely internal) and can wait until 7.1.

### CODE REVIEW

**Comment by dgw** [comment]
This branch is now based on part of #1772, just so the Sphinx build of it will show everything including `PreTrigger` (which is already added to the docs in that nearly-ready-to-merge patch). The first two commits therefore can be ignored when reviewing this.

File: sopel/trigger.py [hunk 1]
```diff
@@ -20,15 +20,82 @@
...
+
+    .. py:attribute:: tags
+
+        Any IRCv3 message tags attached to the line, as a :class:`dict`.
+
+    .. py:attribute:: text
+
+        The last argument of the IRC command.
+
+        If the line contains ``:``, :attr:`text` will be the text following it.
```
[the remark this answers is not in the document]
Comment (dgw) [reply sopel/trigger.py]: I was writing the documentation based on what *should* happen, not necessarily what the code actually *does*. But this should be fixed in the latest push.

**Comment by dgw** [comment]
@Exirel I amended some wording and tacked on a fix for `PreTrigger.text` not always existing. (Marking the review note resolved is for you to do, as a final signal that you're happy with the fix. 😸)

File: sopel/trigger.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Exirel) [reply sopel/trigger.py]: And it is!

**Pull request merged**